### DISCUSSION TITLE
Installing Milvus on Docker on Windows 11

### DISCUSSION THREAD

**Discussion by JKCOkafoeze** [milvus-io/milvus#33937] [Q&A and General discussion]
I'm installing Milvus on docker on Windows 11. Mivus is not connecting on docker as Milvus exits immediately after running. And from jupyter this is the error I get: MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530, illegal connection params or server unavailable)>. I would a lead to be able to establish a connection

**Comment by yhmo** [comment]
Typically, two possibilities of this error:
1. The Milvus container failed to start, double-check the log of milvus, maybe there are some critical errors.
2. Network problem, maybe port 19530 is not exposed in the docker-compose.yaml

**Reply by JKCOkafoeze** [reply]
port 19530, I believe is exposed. See content of the docker-compose.yaml

version: '3.5'
services:
  etcd:
    image: quay.io/coreos/etcd:v3.5.0
    environment:
      - ETCD_AUTO_COMPACTION_RETENTION=1
      - ETCD_QUOTA_BACKEND_BYTES=4294967296
    command: etcd -advertise-client-urls http://etcd:2379 -listen-client-urls http://0.0.0.0:2379 -listen-peer-urls http://0.0.0.0:2380
    volumes:
      - ./volumes/etcd:/etcd
    ports:
      - "2379:2379"
      - "2380:2380"

  minio:
    image: minio/minio:RELEASE.2021-06-17T00-10-46Z
    environment:
      - MINIO_ACCESS_KEY=minioadmin
      - MINIO_SECRET_KEY=minioadmin
    command: server /data
    volumes:
      - ./volumes/minio:/data
    ports:
      - "9000:9000"

  milvus:
    image: milvusdb/milvus:latest
    command: ["milvus", "run", "standalone"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      MINIO_ACCESS_KEY: minioadmin
      MINIO_SECRET_KEY: minioadmin
    volumes:
      - ./volumes/milvus:/milvus
    ports:
      - "19530:19530"

**Comment by xiaofan-luan** [comment]
> I'm installing Milvus on docker on Windows 11. Mivus is not connecting on docker as Milvus exits immediately after running. And from jupyter this is the error I get: MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530, illegal connection params or server unavailable)>. I would a lead to be able to establish a connection

did you use docker or wsl?

**Reply by JKCOkafoeze** [reply]
I use docker

**Comment by xiaofan-luan** [comment]
I believe milvus can not be deployed on windows right now. if using docker, you need to exposed 19530 port on config

**Reply by JKCOkafoeze** [reply]
I believe I exposed the port. See yml file:

version: '3.5'
services:
  etcd:
    image: quay.io/coreos/etcd:v3.5.0
    environment:
      - ETCD_AUTO_COMPACTION_RETENTION=1
      - ETCD_QUOTA_BACKEND_BYTES=4294967296
    command: etcd -advertise-client-urls http://etcd:2379 -listen-client-urls http://0.0.0.0:2379 -listen-peer-urls http://0.0.0.0:2380
    volumes:
      - ./volumes/etcd:/etcd
    ports:
      - "2379:2379"
      - "2380:2380"

  minio:
    image: minio/minio:RELEASE.2021-06-17T00-10-46Z
    environment:
      - MINIO_ACCESS_KEY=minioadmin
      - MINIO_SECRET_KEY=minioadmin
    command: server /data
    volumes:
      - ./volumes/minio:/data
    ports:
      - "9000:9000"

  milvus:
    image: milvusdb/milvus:latest
    command: ["milvus", "run", "standalone"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      MINIO_ACCESS_KEY: minioadmin
      MINIO_SECRET_KEY: minioadmin
    volumes:
      - ./volumes/milvus:/milvus
    ports:
      - "19530:19530"

**Comment by JKCOkafoeze** [comment]
I'm using docker
I exposed the port. See attached yaml file
